Client/Frontend/Browser/FaviconHandler.swift

```diff
@@ -97,6 +101,10 @@ class FaviconHandler {
         favicon.width = Int(image.size.width)
         favicon.height = Int(image.size.height)
         
+        if let tab = tab, !tab.isPrivate {
+          webImageCache.cacheImage(image: image, data: data ?? Data(), url: url)
```

Comment:
`data ?? Data()` fallback is a bit weird, if there's no data wouldn't we have a an empty image? Maybe we should just guard against missing data at the top with the `guard let image = image` part